lib/ansible/modules/cloud/amazon/ec2_transit_gateway_info.py

```diff
@@ -0,0 +1,263 @@
+            type: dict
+            sample:
+              Name: "A sample TGW"
+        transit_gateway_arn:
+            description: The Amazon Resource Name (ARN) of the transit gateway.
+            returned: always
+            type: str
+            sample: "arn:aws:ec2:us-west-2:1234567654323:transit-gateway/tgw-02c42332e6b7da829"
+        transit_gateway_id:
+            description: The ID of the transit gateway.
+            returned: always
+            type: str
+            sample: "tgw-02c42332e6b7da829"
+'''
+
+try:
+    from botocore.exceptions import BotoCoreError, ClientError
+except Exception:
+    pass
+    # handled by imported AnsibleAWSModule
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import (
+    AWSRetry,
+    boto3_tag_list_to_ansible_dict,
+    camel_dict_to_snake_dict,
+    ansible_dict_to_boto3_filter_list
+)
+
+
+class AnsibleEc2TgwInfo(object):
+
+    def __init__(self, module, results):
+        self._module = module
+        self._results = results
+        self._connection = self._module.client('ec2')
+        self._check_mode = self._module.check_mode
+
+        if not hasattr(self._connection, 'describe_transit_gateways'):
+            self._module.fail_json(msg='transit gateway module requires boto3 >= 1.9.52')
+
+    @AWSRetry.exponential_backoff()
+    def describe_transit_gateways(self):
+        """
+        Describe transit gateways.
+
+        module  : AnsibleAWSModule object
+        connection  : boto3 client connection object
+        """
+        # collect parameters
+        filters = ansible_dict_to_boto3_filter_list(self._module.params.get('filters'))
+        transit_gateway_ids = self._module.params.get('transit_gateway_ids')
+
+        # init empty list for return vars
+        transit_gateway_info = list()
+
+        # Get the basic transit gateway info
+        try:
+            response = self._connection.describe_transit_gateways(
+                TransitGatewayIds=transit_gateway_ids, Filters=filters)
+        except (BotoCoreError, ClientError) as e:
+            if e.response['Error']['Code'] == 'InvalidTransitGatewayID.NotFound':
+                self._module.exit_json(transit_gateways=[])
+            else:
+                self._module.fail_json_aws(e)
+
+        for transit_gateway in response['TransitGateways']:
+            transit_gateway_info.append(camel_dict_to_snake_dict(transit_gateway, ignore_list=['Tags']))
+            # convert tag list to ansible dict
+            transit_gateway_info[-1]['tags'] = boto3_tag_list_to_ansible_dict(transit_gateway.get('Tags', []))
+
+        self._module.exit_json(transit_gateways=transit_gateway_info)
+
+
+def setup_module_object():
+    """
+    merge argument spec and create Ansible module object
+    :return: Ansible module object
+    """
+
+    argument_spec = dict(
+        transit_gateway_ids=dict(type='list', default=[], aliases=['transit_gateway_id']),
```

Comment:
```suggestion
        transit_gateway_ids=dict(type='list', default=[], elements='str', aliases=['transit_gateway_id']),
```
Ansible will (from 2.8 on) make sure that the list only contains strings (and nothing else).

Reply:
elements directive added